<h1><b>Waze User Churn Project </b></h1>

<h3>Capstone Project: Google Advanced Data Analytics Professional Certificate</h3>

<h2><b> Background and Business Problem</b></h2>
<p>Waze's navigation app serves millions of drivers worldwide, providing free navigation services through a collaborative community of users, map editors, and partners. However, like many digital platforms, Waze faces the challenge of user churn—users who uninstall the app or stop using it entirely. This creates significant business challenges that directly impact growth and sustainability.

The manual analysis of user behavior patterns was time-consuming and reactive, leading to several critical issues:

- <b>Reactive churn management:</b> Users were already lost before intervention strategies could be implemented
- <b>Inefficient resource allocation:</b> Marketing and retention efforts were spread broadly rather than targeted at high-risk users
- <b>Limited predictive capability:</b> Inability to identify early warning signs of potential churn
- <b>Missed retention opportunities: </b> High-value users at risk of churning went unidentified until it was too late

To address these challenges and support Waze's growth objectives, the data team aims to develop a predictive machine learning model that can identify users at high risk of churning before they actually leave the platform. This proactive approach will enable targeted retention strategies and improve overall user satisfaction.</p>

<h2><b>Business Impact</b></h2>
<p>Implementing a predictive model to identify users at high risk of monthly churn will have direct and measurable impact on Waze's business performance and user retention strategy.</p>
<ul>
    <li><b>Proactie user retention:</b> Early identiifcaton of at-risk users enables targeted intervention before churn occurs. </li>
    <li><b>Optimized vmarketing spend: </b> Resources can be focused on high -risk, high-vvalue users rather than broad based campaigns </li>
    <li><b>Improved user experience: </b>Understanding churn factors helps product teams address pain points that drive users away. </li>
    <li><b>Enhaced revenue potential: </b>Higher retention rates directly correlate with increased user lifetime value and advertising revenu. </li>
    <li><b>Data-driven decision making: </b>Quantifiable insights replace intuition based retetion strategies.</li>
</ul>

<h2><b>Methodology</b></h2>
<h4>1) Data Collection and Preparation </h4>
<ul>
    <li>The goal is to construct a dataframe in Python, perform initial inspection of the provided dataset, and inform  data team members of the findings.</li>
    <li> Understanding the situation: How can the user behaviour data be prepared and organized to identify churn patterns. </li>
    <li> Understand the data: Creating necessary dataframes to understand the user engagement metrics and patterns for future EDA and statistical analysis. </li>
    <li>Understanding variables:  Using insights from the examination of the user activity data to guide deeper investigation into churn prediction variables</li>
    
</ul>


In [1]:
import pandas as pd 
import numpy as np 


In [4]:
df = pd.read_csv("waze_dataset.csv")
df.head(10)

,ID,label,sessions,drives,total_sessions,n_days_after_onboarding,total_navigations_fav1,total_navigations_fav2,driven_km_drives,duration_minutes_drives,activity_days,driving_days,device
0,0,retained,283,226,296.748273,2276,208,0,2628.845068,1985.775061,28,19,Android
1,1,retained,133,107,326.896596,1225,19,64,13715.920550,3160.472914,13,11,iPhone
2,2,retained,114,95,135.522926,2651,0,0,3059.148818,1610.735904,14,8,Android
3,3,retained,49,40,67.589221,15,322,7,913.591123,587.196542,7,3,iPhone
4,4,retained,84,68,168.247020,1562,166,5,3950.202008,1219.555924,27,18,Android
5,5,retained,113,103,279.544437,2637,0,0,901.238699,439.101397,15,11,iPhone
6,6,retained,3,2,236.725314,360,185,18,5249.172828,726.577205,28,23,iPhone
7,7,retained,39,35,176.072845,2999,0,0,7892.052468,2466.981741,22,20,iPhone
8,8,retained,57,46,183.532018,424,0,26,2651.709764,1594.342984,25,20,Android
9,9,churned,84,68,244.802115,2997,72,0,6043.460295,2341.838528,7,3,iPhone


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       14999 non-null  int64  
 1   label                    14299 non-null  object 
 2   sessions                 14999 non-null  int64  
 3   drives                   14999 non-null  int64  
 4   total_sessions           14999 non-null  float64
 5   n_days_after_onboarding  14999 non-null  int64  
 6   total_navigations_fav1   14999 non-null  int64  
 7   total_navigations_fav2   14999 non-null  int64  
 8   driven_km_drives         14999 non-null  float64
 9   duration_minutes_drives  14999 non-null  float64
 10  activity_days            14999 non-null  int64  
 11  driving_days             14999 non-null  int64  
 12  device                   14999 non-null  object 
dtypes: float64(3), int64(8), object(2)
memory usage: 1.5+ MB


In [7]:
df.isna().sum()

ID                           0
label                      700
sessions                     0
drives                       0
total_sessions               0
n_days_after_onboarding      0
total_navigations_fav1       0
total_navigations_fav2       0
driven_km_drives             0
duration_minutes_drives      0
activity_days                0
driving_days                 0
device                       0
dtype: int64

The following can be observed from above:
1) From the first 10 rows we can see that two categorical variables while all other columns are numerical in nature.
2) There are total of 14999 rows and 12 columns. Out of that there are 700 missing values in label column while columns does not have any missing values.
3) The label and device columns are of type "object" while "total_sessions", "driven_km_drives" and "duration_minutes_drives" columns are of type float64 while all other columns are of type int64.